static/js/components/VideoPlayer_test.js

```diff
@@ -1,49 +1,152 @@
+          src: libVideo.getHLSEncodedUrl(video),
+        }
+      ]
     });
+    let enableTouchActivityStub = sandbox.stub();
+    args[2].call({
+      enableTouchActivity: enableTouchActivityStub
+    });
+    sinon.assert.calledWith(enableTouchActivityStub);
+  });
+
+  it('video element is rendered with the correct attributes', () => {
+    let wrapper = renderPlayer();
+    let videoProps = wrapper.find('video').props();
+    assert.equal(videoProps.className, 'video-js vjs-default-skin');
+    assert(videoProps.fluid === undefined);
+    assert(videoProps.controls !== undefined);
   });
 
-  it('renders a component for USwitch video without an iframe', () => {
+  it('video element is rendered with 4 canvas elements when multiangle', () => {
     video.multiangle = true;
     let wrapper = renderPlayer();
-    assert.deepEqual(wrapper.find("USwitchPlayer").props(), {
-      video,
-    });
+    let canvases = wrapper.find('.camera-box');
+    assert.equal(canvases.length, 4);
   });
 
-  it('renders an iframe for playing USwitch video', () => {
+  it('video element is rendered with 1 selected canvas elements when multiangle', () => {
     video.multiangle = true;
-    let wrapper = renderPlayer({ useIframeForUSwitch: true });
-    assert.deepEqual(wrapper.find("iframe").props(), {
-      allowFullScreen: true,
-      className: "uswitch-odl-medium",
-      frameBorder: "0",
-      scrolling: "no",
-      src: makeEmbedUrl(video.key),
-    });
+    let wrapper = renderPlayer();
+    let canvas = wrapper.find('.camera-box-selected').at(0);
+    assert.equal(canvas.props().id, 'upperLeft');
+  });
+
+  it('selected video screen changes on click', () => {
+    video.multiangle = true;
+    let wrapper = renderPlayer();
+    let canvases = wrapper.find('.camera-box');
+    canvases.at(3).prop('onClick')();
+    sinon.assert.calledWith(cornerFunction, 'lowerRight');
+  });
+
+  it('cropVideo modifies style and configureCameras function called', () => {
+    video.multiangle = true;
+    sandbox.stub(window, 'getComputedStyle').returns({maxHeight: 600});
+    let wrapper = renderPlayer();
+    wrapper.instance().player = playerStub;
+    wrapper.instance().videoNode = nodeStub;
+    wrapper.instance().videoContainer = containerStub;
+    wrapper.instance().cropVideo();
+    assert.deepEqual(wrapper.instance().videoNode.style,
+      {
+        "left": "640px",
+        "top": "360px",
+        "OTransform": "scale(2)"
+      }
+    );
+  });
+
+  it('drawCanvas calls inner drawCanvasImage', () => {
+    video.multiangle = true;
+    sandbox.stub(window, 'getComputedStyle').returns({maxHeight: 600});
+    let wrapper = renderPlayer();
+    wrapper.instance().player = playerStub;
+    wrapper.instance().videoNode = nodeStub;
+    let canvas = wrapper.find('.camera-box').at(0);
+    assert.throws(() => wrapper.instance().drawCanvas(canvas, true, false), TypeError, 'getContext');
+  });
+
+  it('subtitles added to player if present', () => {
```

Comment:
Can you add a test where the player has two existing tracks and there are two subtitles coming from `video.videosubtitle_set`, one which is the same as the existing one and one which is new? It would check that we add, remove, and keep existing subtitles appropriately

Reply:
OK